src/l2/XERC20PufferVault.sol

```diff
@@ -0,0 +1,285 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity >=0.8.4 <0.9.0;
+
+import { IXERC20 } from "./interface/IXERC20.sol";
+import { UUPSUpgradeable } from "@openzeppelin-contracts-upgradeable/proxy/utils/UUPSUpgradeable.sol";
+import { AccessManagedUpgradeable } from
+    "@openzeppelin-contracts-upgradeable/access/manager/AccessManagedUpgradeable.sol";
+import { ERC20PermitUpgradeable } from
+    "@openzeppelin-contracts-upgradeable/token/ERC20/extensions/ERC20PermitUpgradeable.sol";
+
+contract XERC20PufferVault is IXERC20, AccessManagedUpgradeable, ERC20PermitUpgradeable, UUPSUpgradeable {
+    /**
+     * @notice The duration it takes for the limits to fully replenish
+     */
+    uint256 private constant _DURATION = 1 days;
+
+    /**
+     * @notice The address of the lockbox contract
+     */
+    address public lockbox;
+
+    /**
+     * @notice Maps bridge address to bridge configurations
+     */
+    mapping(address => Bridge) public bridges;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(address accessManager) public initializer {
+        __AccessManaged_init(accessManager);
+        __ERC20_init("xPufETH", "xPufETH");
+        __ERC20Permit_init("xPufETH");
+    }
+
+    /**
+     * @notice Mints tokens for a user
+     * @dev Can only be called by a bridge
+     * @param _user The address of the user who needs tokens minted
+     * @param _amount The amount of tokens being minted
+     */
+    function mint(address _user, uint256 _amount) public {
```

Comment:
I'm against parameters that have underscores. This function can be external, and we can add `restricted`, but make it public so that we can pause it in case of an emergency.